inject/register.go

```diff
@@ -0,0 +1,20 @@
+package inject
+
+// RegisterType registers the type of the given T as the concrete type to satisfy the service type
+// T when instances are resolved from a [ServiceProvider] built from the given [ServiceCollection].
+// After the instance is resolved, every exported field will be initialized by the same
+// [ServiceProvider]. Note that the given instance of T is not used directly even for types
+// registered with Singleton lifetime.
+func RegisterType[T any](c *ServiceCollection, lifetime ServiceLifetime, type_ T) {}

```

Comment:
May also want a `RegisterInstance` that registers a singleton instance directly and doesn't do additional resolutions for its dependencies. Basically a shorthand for `RegisterFunc` that returns an instance without using the `ServiceResolver` param.